plugins/modules/api.py

```diff
@@ -0,0 +1,292 @@
+    knowledge: "false"
+    location: ""
+    made_sla: "true"
+    notify: "1"
+    number: "INC0010204"
+    opened_at: "2022-07-06 08:53:05"
+    opened_by: "6816f79cc0a8016401c5a33be04be441"
+    order: ""
+    origin_id: ""
+    origin_table: ""
+    parent: ""
+    parent_incident: ""
+    priority: "5"
+    problem_id: ""
+    reassignment_count: "0"
+    reopen_count: "0"
+    reopened_by: ""
+    reopened_time: ""
+    resolved_at: ""
+    resolved_by: ""
+    rfc: ""
+    route_reason: ""
+    service_offering: ""
+    severity: "3"
+    short_description: "my-incident"
+    sla_due: ""
+    state: "1"
+    subcategory: ""
+    sys_class_name: "incident"
+    sys_created_by: "admin"
+    sys_created_on: "2022-07-06 08:53:05"
+    sys_domain: "global"
+    sys_domain_path: "/"
+    sys_id: "35b5fb4197245110949235dfe153af06"
+    sys_mod_count: "0"
+    sys_tags: ""
+    sys_updated_by: "admin"
+    sys_updated_on: "2022-07-06 08:53:05"
+    task_effective_number: "INC0010204"
+    time_worked: ""
+    universal_request: ""
+    upon_approval: "proceed"
+    upon_reject: "cancel"
+    urgency: "3"
+    user_input: ""
+    watch_list: ""
+    work_end: ""
+    work_notes: ""
+    work_notes_list: ""
+    work_start: ""
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+
+from ..module_utils import arguments, client, errors, table
+from ..module_utils.api import (
+    table_name,
+    get_query_by_sys_id,
+    ACTION_POST,
+    ACTION_PATCH,
+    ACTION_DELETE,
+    get_data,
+    sys_id_present,
+)
+
+
+def update_resource(module, table_client):
+    query = get_query_by_sys_id(module)
+    record_old = table_client.get_record(table_name(module), query)
+    if record_old is None:
+        return False, None, dict(before=None, after=None)
+    record_new = table_client.update_record(
+        table_name(module), record_old, get_data(module), module.check_mode
+    )
+    return True, record_new, dict(before=record_old, after=record_new)
+
+
+def create_resource(module, table_client):
+    # At the moment, creating a resource is not idempotent (meaning: If a record with such data as specified in
+    # module.params["data"] already exists, such resource will get created once again).
+    new = table_client.create_record(
+        table=table_name(module),
+        payload=get_data(module),
+        check_mode=module.check_mode,
+    )
+    return True, new, dict(before=None, after=new)
+
+
+def delete_resource(module, table_client):
+    query = get_query_by_sys_id(module)
+    record = table_client.get_record(table_name(module), query)
+    if record is None:
+        return False, None, dict(before=None, after=None)
+    table_client.delete_record(table_name(module), record, module.check_mode)
+    return True, None, dict(before=record, after=None)
+
+
+def run(module, table_client):
+    action = module.params["action"]
+    if (action == ACTION_PATCH or action == ACTION_DELETE) and not sys_id_present(
```

Comment:
This conditional can be omitted since `AnsibleModule` can check that for you if parameter `required_if` is specified. See below.

Reply:
Resolved with https://github.com/ansible-collections/servicenow.itsm/pull/179/commits/79fec49a997d4f3f8dbeb42b4ce1ecba2c838627.